File: Python/even-fibonacci-numbers.py
### Problem 2
### https://projecteuler.net/problem=2

from datetime import datetime
startTime = datetime.now()

fibonacci1 = 1      # previous term in sequence
fibonacci2 = 2      # current term in sequence
fibonacci3 = 0      # placeholder
sum = 0             # sum of even-valued terms

while fibonacci2 < 4000000:
    if fibonacci2 % 2 == 0:
        sum += fibonacci2
    fibonacci3 = fibonacci1 + fibonacci2
    fibonacci1 = fibonacci2
    fibonacci2 = fibonacci3

print(sum)
print(datetime.now() - startTime)

# # # # # # # # # #
# Answer: 4613732 #
# Time: 0:00.001  #
# # # # # # # # # #

File: java/even-fibonacci-numbers.java
// Problem 21
// https://projecteuler.net/problem=2

import java.time.Duration;
import java.time.Instant;
import java.util.*;

class Problem2 {
    public static void main(String[] args) {
        Instant startTime = Instant.now();
        evenSum(4000000);
        System.out.println(Duration.between(startTime, Instant.now()).toString());
    }

    public static void evenSum(int limit) {
        List<Integer> sequence=new ArrayList<Integer>();
        sequence.add(1);
        sequence.add(2);
        int sum = 0;
        while (sequence.get(sequence.size()-1) < limit) {
            sequence.add(sequence.get(sequence.size()-1) + sequence.get(sequence.size()-2));
        }
        if (sequence.get(sequence.size()-1) > limit) {
            sequence.remove(sequence.size()-1);
        }
        for (int i = 0; i < sequence.size(); i++) {
            if (sequence.get(i) % 2 == 0) {
                sum += sequence.get(i);
            }
        }
        System.out.println(sum);
    }
}

/////////////////////
// Answer: 4613732 //
// Time: 0:00.001  //
/////////////////////
